### PR TITLE
Adding support for new internal IP network range resource

### PR DESCRIPTION
**Pull request by cars**
This PR adds support for a new resource type vra_network_ip_range that is attached to fabric networks for managing ranges of IP vRA can assign to VMs when an IPAM solution like Infoblox isn't in play.

### CODE REVIEW

File: vra/resource_network_ip_range.go [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+	createNetworkIPRangeCreated, createNetworkIPRangeAccepted, err := apiClient.NetworkIPRange.CreateInternalNetworkIPRange(network_ip_range.NewCreateInternalNetworkIPRangeParams().WithBody(&networkIPRangeSpecification))
+	if err != nil {
+		return err
+	}
+	if createNetworkIPRangeCreated != nil {
+		d.SetId(*createNetworkIPRangeCreated.Payload.ID)
+	} else {
+		d.SetId(*createNetworkIPRangeAccepted.Payload.ID)
+	}
```
Comment (dmettem) [inline vra/resource_network_ip_range.go:109]: Payload returned by [`createNetworkIPRangeCreated`](https://github.com/vmware/vra-sdk-go/blob/master/pkg/client/network_ip_range/create_internal_network_ip_range_responses.go#L74) and [`createNetworkIPRangeAccepted`](https://github.com/vmware/vra-sdk-go/blob/master/pkg/client/network_ip_range/create_internal_network_ip_range_responses.go#L107) is of type RequestTracker.

You may refer to [`vra_machine`](https://github.com/vmware/terraform-provider-vra/blob/master/vra/resource_machine.go#L248) or other IaaS resources for an example.

File: vra/resource_network_ip_range.go [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+			},
+			"ip_version": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{"IPv4", "IPv6"}, true),
+			},
+			"progress": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
```
Comment (dmettem) [inline vra/resource_network_ip_range.go:55]: `progress` is an attribute from RequestTracker and is not really part of the networkIPRange model. You may omit this.

File: vra/resource_network_ip_range.go [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+		Create: resourceNetworkIPRangeCreate,
+		Read:   resourceNetworkIPRangeRead,
+		Update: resourceNetworkIPRangeUpdate,
+		Delete: resourceNetworkIPRangeDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
```
[the remark this answers is not in the document]
Comment (dmettem) [reply vra/resource_network_ip_range.go:25]: Nit. While you are at addressing this comment, please order the arguments and attributes in ascending order.

**Review by dmettem — CHANGES_REQUESTED** [review]
Thanks @cars for the contribution and the PR.

Most of the comments are nit. Please update the PR once the changes are complete.

File: vra/resource_network_ip_range.go (hunk 1 above)
Comment (cars) [reply vra/resource_network_ip_range.go:109]: @dmettem  - I agree the swagger says it's supposed to return a response of type RequestTracker, but I'm not seeing that in practice at least not on-prem ....I don't have access to vRA Cloud to validate if it behaves differently or not, so wondering if someone else can. I've uploaded the relevant snippet from a log to https://gist.github.com/cars/83b6b4955a2400927f02a09676607da6#file-tf_network_ip_range-log 

The POST starts  [here](https://gist.github.com/cars/83b6b4955a2400927f02a09676607da6#file-tf_network_ip_range-log-L21)  And the non-RequestTracker response is [here](https://gist.github.com/cars/83b6b4955a2400927f02a09676607da6#file-tf_network_ip_range-log-L52)

I wondering if maybe there's an error in the swagger or ... ?

Comment (dmettem) [reply vra/resource_network_ip_range.go:109]: @cars - Sorry for late response. I'll verify this on 8.1 build and get back.

Comment (dmettem) [reply vra/resource_network_ip_range.go:109]: @cars - From the snippet you provided, I see that you are not getting the RequestTracker in the response. I opened an internal issue to correct the swagger. 

For now, you can ignore this comment, and work on the rest to merge into master. Thank You!

File: go.mod [hunk 4]
```diff
@@ -7,4 +7,6 @@ require (
 	github.com/go-openapi/strfmt v0.19.5
 	github.com/hashicorp/terraform-plugin-sdk v1.9.1
 	github.com/vmware/vra-sdk-go v0.2.10
+	golang.org/x/lint v0.0.0-20200302205851-738671d3881b // indirect
+	golang.org/x/tools v0.0.0-20200904185747-39188db58858 // indirect
```
[the remark this answers is not in the document]
Comment (cars) [reply go.mod]: removed them (I'm blaming vscode)

**Comment by dmettem** [comment]
@cars - Please revert my last commit, rebase master and update the PR.

I couldn't push to this branch from my local to update the PR to sign my last commit. Thanks!

**Review by dmettem — APPROVED** [review]

**Pull request merged**